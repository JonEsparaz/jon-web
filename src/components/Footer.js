import React from 'react';

function Footer() {
  return (
    <div className="FooterContainer">
      <span><a href="https://twitter.com/JonEsparaz" rel="noopener noreferrer" target="_blank"><img className="FooterSocialImg" style={{ marginRight: "0.5vw", marginLeft: "3vw", marginTop: "2.5vw", marginBottom: "2.5vw", height: "30px"}} src="twitter.svg" alt="Twitter Logo" /></a></span>
      <span><a href="https://github.com/JonEsparaz" rel="noopener noreferrer" target="_blank"><img className="FooterSocialImg" style={{ marginRight: "0.5vw", marginLeft: "3vw", marginTop: "2.5vw", marginBottom: "2.5vw", height: "30px"}} src="github.png" alt="GitHub Logo" /></a></span>
      <span><a href="https://www.youtube.com/c/JonEsparaz" rel="noopener noreferrer" target="_blank"><img className="FooterSocialImg" style={{ marginRight: "0.5vw", marginLeft: "3vw", marginTop: "2.5vw", marginBottom: "2.5vw", height: "30px"}} src="youtube.svg" alt="YouTube Logo" /></a></span>
      <span><a href="https://www.linkedin.com/in/jonathanesparaz/" rel="noopener noreferrer" target="_blank"><img className="FooterSocialImg" style={{ marginRight: "0.5vw", marginLeft: "3vw", marginTop: "2.5vw", marginBottom: "2.5vw", height: "30px"}} src="linkedin-256.png" alt="LinkedIn Logo" /></a></span>

    </div>
  );
}

export default Footer;
