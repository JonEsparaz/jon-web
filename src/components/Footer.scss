@import '../theme.scss';

.FooterContainer {
  width: 100%;
  text-align: center;
  padding: 40px;
  bottom: 0;
  font-family: $font;
}

.footer-icon {
  width: 32px;
  height: 32px;
}

@media (min-width: 768px) {
  .footer-link {
    margin: 0 1.3vw;
    transition: opacity 0.3s;
    opacity: 1;
    &:hover {
      opacity: 0.6;
    }
  }
}

@media (max-width: 767px) {
  .footer-link {
    margin: 0 3vw;
  }
}
