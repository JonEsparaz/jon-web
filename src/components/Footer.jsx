import React from 'react';
import TwitterIcon from '@material-ui/icons/Twitter';
import GitHubIcon from '@material-ui/icons/GitHub';
import YouTubeIcon from '@material-ui/icons/YouTube';
import LinkedInIcon from '@material-ui/icons/LinkedIn';
import './Footer.scss';

function Footer() {
  return (
    <div className="FooterContainer">
      <span><a href="https://twitter.com/JonEsparaz" rel="noopener noreferrer" target="_blank"><TwitterIcon fontSize="large" style={{ marginRight: "1.5vw", marginLeft: "1.5vw", color: "#212529"}}/></a></span>
      <span><a href="https://github.com/JonEsparaz" rel="noopener noreferrer" target="_blank"><GitHubIcon fontSize="large" style={{ marginRight: "1.5vw", marginLeft: "1.5vw", color: "#212529"}}/></a></span>
      <span><a href="https://www.youtube.com/c/JonEsparaz" rel="noopener noreferrer" target="_blank"><YouTubeIcon fontSize="large" style={{ marginRight: "1.5vw", marginLeft: "1.5vw", color: "#212529"}}/></a></span>
      <span><a href="https://www.linkedin.com/in/jonathanesparaz/" rel="noopener noreferrer" target="_blank"><LinkedInIcon fontSize="large" style={{ marginRight: "1.5vw", marginLeft: "1.5vw", color: "#212529"}}/></a></span>
      <p style={{marginTop:"20px"}}> &copy; {new Date().getFullYear()} Jonathan Esparaz</p>
    </div>
  );
}

export default Footer;
