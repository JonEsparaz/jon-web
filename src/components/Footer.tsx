import React from 'react';
import TwitterIcon from '@material-ui/icons/Twitter';
import GitHubIcon from '@material-ui/icons/GitHub';
import YouTubeIcon from '@material-ui/icons/YouTube';
import LinkedInIcon from '@material-ui/icons/LinkedIn';
import './Footer.scss';

export default function Footer(): JSX.Element {
  return (
    <div className="FooterContainer">
      <a
        aria-label="Twitter"
        href="https://twitter.com/JonEsparaz"
        rel="noopener noreferrer"
        target="_blank"
        className="footer-link"
      >
        <TwitterIcon fontSize="large" style={{ color: 'black' }} />
      </a>
      <a
        aria-label="GitHub"
        href="https://github.com/JonEsparaz"
        rel="noopener noreferrer"
        target="_blank"
        className="footer-link"
      >
        <GitHubIcon fontSize="large" style={{ color: 'black' }} />
      </a>
      <a
        aria-label="YouTube"
        href="https://www.youtube.com/c/JonEsparaz"
        rel="noopener noreferrer"
        target="_blank"
        className="footer-link"
      >
        <YouTubeIcon fontSize="large" style={{ color: 'black' }} />
      </a>
      <a
        aria-label="LinkedIn"
        href="https://www.linkedin.com/in/jonathanesparaz/"
        rel="noopener noreferrer"
        target="_blank"
        className="footer-link"
      >
        <LinkedInIcon fontSize="large" style={{ color: 'black' }} />
      </a>
      <div style={{ marginTop: '20px', color: 'black' }}>
        {' '}
        &copy; {new Date().getFullYear()} Jonathan Esparaz
      </div>
    </div>
  );
}
