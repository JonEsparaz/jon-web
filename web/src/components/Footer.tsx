import React from 'react';
import './Footer.scss';

export default function Footer(): JSX.Element {
  const links = [
    {
      label: 'Twitter',
      link: 'https://twitter.com/JonEsparaz',
      icon: '/svg/twitter.svg',
    },
    {
      label: 'GitHub',
      link: 'https://github.com/JonEsparaz',
      icon: '/svg/github.svg',
    },
    {
      label: 'LinkedIn',
      link: 'https://www.linkedin.com/in/jonathanesparaz/',
      icon: '/svg/linkedin.svg',
    },
    {
      label: 'YouTube',
      link: 'https://www.youtube.com/c/JonEsparaz',
      icon: '/svg/youtube.svg',
    },
  ];

  return (
    <div className="container mt-5 mb-4 text-center">
      {links.map((item) => {
        return (
          <a
            key={item.label}
            aria-label={item.label}
            href={item.link}
            rel="noopener noreferrer"
            target="_blank"
            className="footer-link"
          >
            <img src={item.icon} alt="" width={32} height={32} />
          </a>
        );
      })}
      <p className="mt-4">&copy; {new Date().getFullYear()} Jonathan Esparaz</p>
    </div>
  );
}
