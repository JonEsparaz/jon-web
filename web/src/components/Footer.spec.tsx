import React from 'react';
import { render } from '@testing-library/react';
import Footer from './Footer';

const labels = ['Twitter', 'GitHub', 'YouTube', 'LinkedIn'];

test('footer spanshot, links', () => {
  const { container, getAllByRole, getByLabelText } = render(<Footer />);

  const links = getAllByRole('link');

  // four links
  expect(links.length).toBe(4);

  // each link matches label
  for (let i = 0; i < 4; i++) {
    expect(getByLabelText(labels[i])).toBeTruthy();
  }

  expect(container.innerHTML).toMatchSnapshot();
});
