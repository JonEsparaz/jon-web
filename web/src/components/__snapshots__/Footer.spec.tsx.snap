// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer spanshot, links 1`] = `"<div class=\\"container mt-5 mb-4 text-center\\"><a aria-label=\\"Twitter\\" href=\\"https://twitter.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/twitter.svg\\" alt=\\"\\" width=\\"32\\" height=\\"32\\"></a><a aria-label=\\"GitHub\\" href=\\"https://github.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/github.svg\\" alt=\\"\\" width=\\"32\\" height=\\"32\\"></a><a aria-label=\\"LinkedIn\\" href=\\"https://www.linkedin.com/in/jonathanesparaz/\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/linkedin.svg\\" alt=\\"\\" width=\\"32\\" height=\\"32\\"></a><a aria-label=\\"YouTube\\" href=\\"https://www.youtube.com/c/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><img src=\\"/svg/youtube.svg\\" alt=\\"\\" width=\\"32\\" height=\\"32\\"></a><p class=\\"mt-4\\">Â© 2021 Jonathan Esparaz</p></div>"`;
